plugin/backend_linux.go

```diff
@@ -168,68 +178,163 @@ func computePrivileges(pd distribution.PullData) (types.PluginPrivileges, error)
 	return privileges, nil
 }
 
-// Privileges pulls a plugin config and computes the privileges required to install it.
-func (pm *Manager) Privileges(name string, metaHeader http.Header, authConfig *types.AuthConfig) (types.PluginPrivileges, error) {
-	_, pd, err := pm.pull(name, metaHeader, authConfig)
+// parseRemoteRef parses the remote reference into a reference.Named
+// returning the tag associated with the reference. In the case the
+// given reference string includes both digest and tag, the returned
+// reference will have the digest without the tag, but the tag will
+// be returned.
+func parseRemoteRef(remote string) (reference.Named, string, error) {
+	// Parse remote reference, supporting remotes with name and tag
+	// NOTE: Using distribution reference to handle references
+	// containing both a name and digest
+	remoteRef, err := distreference.ParseNamed(remote)
```

Comment:
We end up calling with both, we use the `WithName` later on from the docker reference package which will normalize. It is a completely mess, but need the tag and digest support from the real reference package and the normalization from the docker one.